### PR TITLE
Update docs & add requirements for Scorecard

### LINKED ISSUE #67: Clarify Scorecard requirements in docs
We've recently been exploring the use of OpenSSF Scorecard Monitor for our various open source organizations, but hit a roadbump after finding that Scorecard Monitor was only discovering a handful of repositories in our orgs (regardless of which projects were listed in `scope.json`). 

After some investigation, I realized that OpenSSF was [automatically tracking](https://github.com/ossf/scorecard/blob/main/docs/faq.md#can-i-preview-my-projects-score) a select group of our projects upstream, which is why some projects already scores and others didn't. I'd like to update the README docs to clarify these requirements, and provide some hints in case others experience similar issues.

### PR DESCRIPTION
**Pull request by lelia**
Fixes #67.

In addition to clarifying requirements for Scorecard and offering some hints, this PR also fixes some Markdown lint violations, uses more consistent casing, and slightly revises some phrasing. Feedback welcome!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,12 @@
 
 If you're feeling overwhelmed by an avalanche of repository scorecards in your organization, you can breathe easy: Automation is here to make your life easier! It will streamline the process of keeping track of them all by providing a comprehensive report in Markdown and a local database in JSON with all the scores. Furthermore, to stay on top of any changes in the scores, you can choose to get notifications through Github Issues.
 
+## ✅ Requirements
+
+Please ensure that any repository you wish to track with Scorecard Monitor has already been analyzed by [OpenSSF Scorecard](https://github.com/ossf/scorecard) at least once. This can be accomplished using the official [GitHub Action](https://github.com/ossf/scorecard-action) or the [Scorecard CLI](https://github.com/ossf/scorecard?tab=readme-ov-file#scorecard-command-line-interface). It's also possible that some repositories in your organization are already being [automatically tracked](https://github.com/ossf/scorecard/blob/main/docs/faq.md#can-i-preview-my-projects-score) by OpenSSF via this [CSV file](https://github.com/ossf/scorecard/blob/main/cron/internal/data/projects.csv)!
```
Comment (UlisesGascon) [inline README.md:11]: I discovered recently that the repositories that are automatically tracked does not include certain categories in the analysis (CI-Tests, Contributors, Dependency-Update-Tool, Webhooks). Do you think that we should mention it?

Ref: https://github.com/ossf/scorecard/issues/3438#issuecomment-1701426865

**Review by UlisesGascon — APPROVED** [review]
Thanks @lelia. This is a great contribution, and it will help a lot in the onboarding process!

File: README.md (hunk 1 above)
Comment (lelia) [reply README.md:11]: That's a good catch - happy to tweak the phrasing to mention that if you'd like me to file a separate PR!

Comment (UlisesGascon) [reply README.md:11]: Yes, feel free to create a pr for it :)

Comment (lelia) [reply README.md:11]: [Done](https://github.com/UlisesGascon/openssf-scorecard-monitor/pull/73)!

**Pull request merged**